src/qibo/tests/test_models_error_mitigation.py

```diff
@@ -0,0 +1,180 @@
+import numpy as np
+import pytest
+
+import qibo
+from qibo import gates
+from qibo.hamiltonians import SymbolicHamiltonian
+from qibo.models import Circuit
+from qibo.models.error_mitigation import CDR, ZNE, sample_training_circuit, vnCDR
+from qibo.noise import DepolarizingError, NoiseModel
+from qibo.symbols import Z
+
+
+def get_noise_model(error):
+    noise = NoiseModel()
+    noise.add(error, gates.CNOT)
+    return noise
+
+
+@pytest.mark.parametrize("nqubits", [3])
+@pytest.mark.parametrize(
+    "noise",
+    [get_noise_model(DepolarizingError(0.1))],
+)
+@pytest.mark.parametrize("solve", [False, True])
+def test_zne(backend, nqubits, noise, solve):
+    """Test that ZNE reduces the noise."""
+    qibo.set_threads(1)
```

Comment:
Thanks for setting the threads, I believe the tests are faster now, do you also observe some difference? 

One small detail, it may be better to use
```suggestion
    backend.set_threads(1)
```
so that we only set the threads for the local backend and we don't affect other tests that run after this.

Reply:
Thanks. By using `backend.set_threads(1)` the times drop from 1min 35sec to 0.5sec for ZNE, from 4min 12sec to 1.9sec for CDR and from 12min 36sec to 3.9sec for vnCDR.